src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -164,15 +167,21 @@ public boolean publish(String message, JSONArray attachments, String color) {
                 if (StringUtils.isNotEmpty(message)) {
                     json.put("text", message);
                 }
-                json.put("attachments", attachments);
+                if (attachments.size() > 0) {
+                    json.put("attachments", attachments);
+                }
                 json.put("link_names", "1");
+                json.put("unfurl_links", "true");
+                json.put("unfurl_media", "true");
 
                 nvps.add(new BasicNameValuePair("payload", json.toString()));
             } else {
                 url = "https://slack.com/api/chat.postMessage?token=" + populatedToken +
                         "&channel=" + roomId.replace("#", "") +
                         "&link_names=1" +
-                        "&as_user=true";
+                        "&as_user=true" +
+                        "&unfurl_links=true" +
```

Comment:
After seeing this again, I was like WAIT A SECOND.. :laughing: 
You do know this very same endpoint accepts JSON?

https://api.slack.com/methods/chat.postMessage
> Accepted content types: application/x-www-form-urlencoded, application/json

Why are we building a URL when we could build JSON?